test/direct_uploads_controller_test.rb

```diff
@@ -27,7 +27,7 @@ class ActiveStorage::DirectUploadsControllerTest < ActionController::TestCase
 
       details = JSON.parse(@response.body)
 
-      assert_match /rails-activestorage\.s3.amazonaws\.com/, details["url"]
+      assert_match /#{SERVICE_CONFIGURATIONS[:s3][:bucket]}\.s3.amazonaws\.com/, details["url"]
```

Comment:
This regular expression fails on my configuration `https://activestorage-test.s3.eu-central-1.amazonaws.com....`, you are missing the part for the region if present
`https://<bucket>.s3.<region>.amazonaws.com....`

Also if we are going to introduce the config hash into the regex than we should also do it [here](https://github.com/rails/activestorage/blob/master/test/service/s3_service_test.rb#L41)